bears/general/AnnotationBear.py

```diff
@@ -28,104 +27,212 @@ def run(self, filename, file, language: str, coalang_dir: str = None):
+                filename, AbsolutePosition(file, position))
+            raise NoCloseError(annotation_start, _range)
+
+        return SourceRange.from_absolute_position(
+            filename,
+            AbsolutePosition(file, position),
+            AbsolutePosition(file, end_end)), end_end
+
+    @staticmethod
+    def get_singleline_strings(file, filename,
+                               text,
+                               string_start, string_end,
+                               position):
+        """
+        Get sourcerange of a single-line string and its end position.
+
+        :param file:
+            A tuple of strings, with each string being a line in the file.
+        :param filename:
+            The name of the file.
+        :param string_start:
+            The string which specifies how a string starts.
+        :param string_end:
+            The string which specifies how a string ends.
+        :position:
+            An integer identifying the position where the string started.
+        :return:
+            A SourceRange object identifying the range of the single-line
+            string and the end_position of the string as an integer.
+        """
+        end_position = text.find(string_end,
+                                 position + 1) + len(string_end) - 1
+        newline = text.find("\n", position + 1)
+        if newline == -1:
+            newline = len(text)
+        if end_position == -1:
+            _range = SourceRange.from_absolute_position(
+                filename, AbsolutePosition(file, position))
+            raise NoCloseError(string_start, _range)
+        if newline > end_position:
+            return SourceRange.from_absolute_position(
+                filename,
+                AbsolutePosition(file, position),
+                AbsolutePosition(file, end_position)), end_position
+
+    @staticmethod
+    def get_singleline_comment(file, filename, text, comment, position):
+        """
+        Get Sourcerange of a single-line comment where the start is the
+        start of comment and the end is the end of line.
+
+        :param file:
+            A tuple of strings, with each string being a line in the file.
+        :param filename:
+            The name of the file.
+        :param comment:
+            The string which specifies the comment.
+        :position:
+            An integer identifying the position where the string started.
+        :return:
+            A SourceRange object identifying the range of the single-line
+            comment and the end_position of the comment as an integer.
+        """
+        end_position = text.find("\n", position + 1)
+        if end_position == -1:
+            end_position = len(text) - 1
+        return SourceRange.from_absolute_position(
+            filename,
+            AbsolutePosition(file, position),
+            AbsolutePosition(file, end_position)), end_position
+
+
+class NoCloseError(Exception):
```

Comment:
Is this really necessary? I don't necessarily like the idea to define a custom Exception inside a Bear. It is good practice to only `try: ... except:` a small subset of applicable errors and a custom exception might slip. Is this any better than just raising a ValueError? 

> Value Error:
"Raised when a built-in operation or function receives an argument that has the right type but an inappropriate value, and the situation is not described by a more precise exception such as IndexError."